mne/dipole.py

```diff
@@ -211,6 +215,107 @@ def __len__(self):
         """Handle len function"""
         return self.pos.shape[0]
 
+    def prep_forward_simulation(self, info, bem, trans=None, min_dist=None,
+                                model_meg=True, model_eeg=True,
+                                n_job=1, verbose=None):
+        """Prepare to project dipoles to sensor-space by creating a discrete
+        source space corresponding to the position of each dipole, and
+        calculate a forward operator.
+
+        ADD MORE INFORMATION HERE
+
+        Parameters
+        ----------
+        info : dict
+            Sensor-info etc., e.g., from real evoked file.
+        bem : str | dict
+            The BEM filename (str) or a loaded sphere model (dict).
+        trans : str | None
+            The head<->MRI transform filename. Must be provided unless BEM
+            is a sphere model.
+        min_dist : float
+            Minimum distance (in milimeters) from the dipole to the inner skull.
+        model_meg : bool
+            Include MEG channels (with or without compensation) in model.
+        model_eeg : bool
+            Include EEG channels in model.
+        n_jobs : int
+            Number of jobs to run in parallel (used in field computation
+            and fitting).
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+
+        Returns
+        -------
+        stc : instance of VolSourceEstimate
+            The dipoles converted to a discrete set of points and associated
+            time courses.
+        fwd : instance of Forward
+            The forward solution.
+
+        See Also
+        --------
+        mne.simulation.simulate_evoked
+
+        Notes
+        -----
+        .. versionadded:: 0.XXXX.0
+        """
+
+        # Convert positions to discrete source space (NB allows duplicates)
+        pos = dict(rr=self.pos, nn=self.ori)
+        sp = _make_discrete_source_space(pos)
+        # What the heck are working_dir and command_line for?!?
+        src = SourceSpaces([sp], dict(working_dir='None', command_line='None'))
+
+        # NB very confused here... First I though Dipoles are in HEAD coords,
```

Comment:
Excellent. Please add comments to clarify this, at least in the `Dipole` class. If you have time, feel free to add some to the `forward` code, too, where it would have been helpful for you. You're at least the third person to work hard to understand the transforms in there :)